pkg/tenant/db/members.go

```diff
@@ -52,11 +53,73 @@ func (m *Member) UnmarshalValue(data []byte) error {
 	return msgpack.Unmarshal(data, m)
 }
 
-// CreateMember adds a new Member to the database.
+func (m *Member) Validate() error {
+	memberName := m.Name
+
+	if strings.ContainsAny(string(memberName[0]), "0123456789") {
+		return ErrNumberFirstCharacter
+	}
+
+	if strings.ContainsAny(memberName, " !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~") {
+		return ErrSpecialCharacters
+	}
+
+	return nil
+}
+
+// Validates data in the member model when a TenantID is required.
+// Ex. CreateTenantProject and UpdateProject
+func (m *Member) ValidateWithID() error {
```

Comment:
We want to have a `TenantID` in the models, but this then causes errors with the server-side `CreateMember` and `CreateProject` handlers. I created a `ValidateID` method that only checks for the ID to eliminate the redundant code. Although, this might not exactly be the most optimal solution at the moment.

Reply:
@pdeziel For now validation will only be applied to `TenantCreateMember`, `TenantCreateProject`, and `ProjectTopicCreate`.